Sources/NIOHTTP2/HTTP2FrameParser.swift

```diff
@@ -0,0 +1,738 @@
+//
+//  HTTP2NativeParser.swift
+//  NIOHTTP2
+//
+//  Created by Jim Dovey on 11/14/18.
+//
+
+import NIO
+import NIOHPACK
+
+/// A temporary item used to hold onto the HPACK encoder/decoder. Something in the real stream state engine will
+/// take its place soon.
+class HTTP2StreamData {
```

Comment:
HPACK encoders and decoders are not per-stream, they are per-connection. We don't need this: the `FrameDecoder` can simply own a `HPACKDecoder`, and the `FrameEncoder` can simply own a `HPACKEncoder`.

Reply:
Oh. Weird, I didn't realize header compression could be re-used across different streams.